### PR TITLE
Fix sometimes-failing LARS tests

### PR DESCRIPTION
**Pull request by rcurtin**
I noticed that the tests for LARS were failing more than once every hundred trials with different random seeds, so I started digging in to see what was wrong.  I came out with a few conclusions:

 * With `float` as an element type, the `PredictTest` can be more flaky---this is expected due to less precision.  So now it allows multiple trials when the element type is `float`.
 
 * When a singularity is encountered, we try to remove the most-recently-added variable from the active set.  However, the `activeGramMatrix` was not removing its columns correctly.
 
 * When computing the signs of correlations, if the correlation was exactly 0, then there was a division by 0.  I fixed that just by adding a small value to the denominator.
 
 * The check to see if an externally-passed Gram matrix satisfied the expected conditions of mean-centeredness and unit-variance was more trouble than it was worth, and I think the note in the documentation is good enough.  So I removed the check, since it does not seem to be reliable.

Then, I ran `[LARSTest]` 1000 times with different random seeds and encountered no failures, so, hopefully this should help with test robustness.

### CODE REVIEW

File: src/mlpack/methods/lars/lars_impl.hpp [hunk 1]
```diff
@@ -794,9 +772,11 @@ LARS<ModelMatType>::Train(const MatType& matX,
         // need to take a step with the previous beta direction towards the next
         // variable we will add.
         s = s.subvec(0, activeSet.size() - 1); // Drop last element.
+        matGramActive = matGramActive.submat(0, 0, activeSet.size() - 1,
+            activeSet.size() - 1);
         matS = s * ones<MatType>(1, activeSet.size());
         // This worked last iteration, so there can't be a singularity.
-        solve(unnormalizedBetaDirection,
+        const bool status2 = solve(unnormalizedBetaDirection,
```
Comment (conradsnicta) [inline src/mlpack/methods/lars/lars_impl.hpp:779]: @rcurtin `status2` doesn't seem to be used anywhere

Comment (rcurtin) [reply src/mlpack/methods/lars/lars_impl.hpp]: Thanks!  Removed in c8d40be8.

**Review by conradsnicta — APPROVED** [review]

File: src/mlpack/tests/sparse_coding_test.cpp [hunk 2]
```diff
@@ -8,10 +8,6 @@
  * 3-clause BSD license along with mlpack.  If not, see
  * http://www.opensource.org/licenses/BSD-3-Clause for more information.
  */
-
-// Note: We don't use BOOST_REQUIRE_CLOSE in the code below because we need
-// to use FPC_WEAK, and it's not at all intuitive how to do that.
-
```
Comment (shrit) [inline src/mlpack/tests/sparse_coding_test.cpp:14]: Nice catch

File: src/mlpack/tests/lars_test.cpp [hunk 3]
```diff
@@ -1094,7 +1225,7 @@ TEMPLATE_TEST_CASE("LARSSelectBetaTest", "[LARSTest]", arma::fmat, arma::mat)
   typedef TestType MatType;
   typedef typename MatType::elem_type ElemType;
 
-  const ElemType tol = (std::is_same<ElemType, double>::value) ? 1e-5 : 1e-3;
+  const ElemType tol = (std::is_same<ElemType, double>::value) ? 1e-5 : 5e-3;
```
Comment (shrit) [inline src/mlpack/tests/lars_test.cpp:1228]: This is a nice way to make the usage of `std::is_same` in a condition with out the need for `constexpr`

**Review by shrit — APPROVED** [review]

**Pull request merged**